dubbo-remoting/dubbo-remoting-netty4/src/main/java/org/apache/dubbo/remoting/transport/netty4/NettyPortUnificationServer.java

```diff
@@ -138,9 +139,31 @@ protected void initChannel(SocketChannel ch) throws Exception {
             bindIp = ANYHOST_VALUE;
         }
         InetSocketAddress bindAddress = new InetSocketAddress(bindIp, bindPort);
-        ChannelFuture channelFuture = bootstrap.bind(bindAddress);
-        channelFuture.syncUninterruptibly();
-        channel = channelFuture.channel();
+        Throwable lastError = null;
+        int retryTimes = getUrl().getParameter(Constants.BIND_RETRY_TIMES, 10);
+        int retryInterval = getUrl().getParameter(Constants.BIND_RETRY_INTERVAL, 3000);
+        for (int i = 0; i < retryTimes; i++) {
```

Comment:
Why would we need binding retry, what's the problem with throwing exception directly on binding failure.

Reply:
Such 20880, 22222... are in `ip_local_port_range` and can be used as a tcp client port. If there is someone requesting remoting using 20880 as client port(allocated by kernel), Dubbo will bind failed.